gadget/gadget_test.go

```diff
@@ -1428,7 +1428,8 @@ func (s *gadgetYamlTestSuite) TestValidateStructureReservedLabels(c *C) {
 		role, label, err string
 	}{
 		{label: "ubuntu-seed", err: `label "ubuntu-seed" is reserved`},
-		{label: "ubuntu-boot", err: `label "ubuntu-boot" is reserved`},
+		// 2020-12-02: disable for customer hotfix
```

Comment:
it would be good to add unit tests with the gadget.yaml you have in the overlord unit tests to here as well, something like this:

```patch
diff --git a/gadget/gadget_test.go b/gadget/gadget_test.go
index f8c5ebed53..6b15a1bfb4 100644
--- a/gadget/gadget_test.go
+++ b/gadget/gadget_test.go
@@ -116,6 +116,44 @@ volumes:
           - image: u-boot.imz
 `)
 
+var mockHybrid18To20GadgetYaml = []byte(`
+volumes:
+  hybrid:
+    bootloader: grub
+    structure:
+      - name: mbr
+        type: mbr
+        size: 440
+        content:
+          - image: pc-boot.img
+      - name: BIOS Boot
+        type: DA,21686148-6449-6E6F-744E-656564454649
+        size: 1M
+        offset: 1M
+        offset-write: mbr+92
+        content:
+          - image: pc-core.img
+      - name: EFI System
+        type: EF,C12A7328-F81F-11D2-BA4B-00A0C93EC93B
+        filesystem: vfat
+        filesystem-label: system-boot
+        size: 1200M
+        content:
+          - source: grubx64.efi
+            target: EFI/boot/grubx64.efi
+          - source: shim.efi.signed
+            target: EFI/boot/bootx64.efi
+          - source: mmx64.efi
+            target: EFI/boot/mmx64.efi
+          - source: grub.cfg
+            target: EFI/ubuntu/grub.cfg
+      - name: Ubuntu Boot
+        type: 0FC63DAF-8483-4772-8E79-3D69D8477DE4
+        filesystem: ext4
+        filesystem-label: ubuntu-boot
+        size: 750M
+`)
+
 var mockClassicGadgetYaml = []byte(`
 defaults:
   system:
@@ -849,6 +887,16 @@ func (s *gadgetYamlTestSuite) TestReadGadgetYamlPCHappy(c *C) {
 	}
 }
 
+func (s *gadgetYamlTestSuite) TestReadHybridGadgetYamlHappy(c *C) {
+	err := ioutil.WriteFile(s.gadgetYamlPath, mockHybrid18To20GadgetYaml, 0644)
+	c.Assert(err, IsNil)
+
+	for _, constraints := range classicModelConstraints {
+		_, err = gadget.ReadInfo(s.dir, constraints)
+		c.Assert(err, IsNil)
+	}
+}
+
 func (s *gadgetYamlTestSuite) TestReadGadgetYamlRPiHappy(c *C) {
 	err := ioutil.WriteFile(s.gadgetYamlPath, gadgetYamlRPi, 0644)
 	c.Assert(err, IsNil)

```

Reply:
something like that will go a in a follow-up